src/DotLottie.tsx

```diff
@@ -0,0 +1,398 @@
+import {
+  forwardRef,
+  useCallback,
+  useImperativeHandle,
+  useRef,
+  type MutableRefObject,
+} from 'react';
+import {
+  requireNativeComponent,
+  UIManager,
+  Platform,
+  type ViewStyle,
+  findNodeHandle,
+} from 'react-native';
+import type { Mode } from './constants';
+import { parseSource } from './utils';
+
+const LINKING_ERROR =
+  `The package 'dotlottie-react-native' doesn't seem to be linked. Make sure: \n\n` +
+  Platform.select({ ios: "- You have run 'pod install'\n", default: '' }) +
+  '- You rebuilt the app after installing the package\n' +
+  '- You are not using Expo Go\n';
+
+export type Dotlottie = {
+  play: () => void;
+  pause: () => void;
+  stop: () => void;
+  setLoop: (value: boolean) => void;
+  setSpeed: (value: number) => void;
+  setPlayMode: (mode: Mode) => void;
+  setFrame: (frame: number) => void;
+  freeze: () => void;
+  unfreeze: () => void;
+  startStateMachine: () => void;
+  stopStateMachine: () => void;
+  loadStateMachine: (stateMachineId: string) => void;
+  setPostEvent: (event: string) => void;
+  addStateMachineEventListener: () => void;
+  removeStateMachineEventListener: () => void;
+  resize: (width: number, height: number) => void;
+  setSegment: (start: number, end: number) => void;
+  setMarker: (marker: string) => void;
+  loadTheme: (themeId: string) => void;
+  loadAnimation: (animationId: string) => void;
+  setManifest: () => void;
```

Comment:
@shamsudeeen-yusuf I don't think that there is a native method called setManifest